### PR TITLE
[WIP] coverage: try timid=1 (PyTracer)

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
/cc @nedbat (via IRC already).
Closing for now.

**Comment by nicoddemus** [comment]
@blueyed can you give a quick summary of what's the purpose of this PR?

**Comment by blueyed** [comment]
Trying out timid=1 (PyTracer).

**Comment by blueyed** [comment]
I plan to minkeypatch `sys.settrace` to always call the coverage tracer, and thought that it might be easier with PyTracer (but probably not).

File: testing/code/test_excinfo.py [hunk 1]
```diff
@@ -238,7 +238,10 @@ def f(n):
                 n += 1
             f(n)
 
+        # Restore trace function (https://bugs.python.org/issue36474).
+        oldtrace = sys.gettrace()
         excinfo = pytest.raises(RuntimeError, f, 8)
+        sys.settrace(oldtrace)
```
Comment (blueyed) [inline testing/code/test_excinfo.py]: Do you think it makes sense to let `pytest.raises` handle this, maybe optionally by default only?

Comment (blueyed) [reply testing/code/test_excinfo.py]: The problem here is that Python forgets about the trace function with RecursionErrors (https://bugs.python.org/issue36474).

**Pull request closed without merging**